apps/dotcom/src/components/DocumentName/DocumentName.tsx

```diff
@@ -0,0 +1,295 @@
+import {
+	Button,
+	DropdownMenu,
+	Kbd,
+	OfflineIndicator,
+	TLDocument,
+	track,
+	useActions,
+	useBreakpoint,
+	useEditor,
+	useTranslation,
+} from '@tldraw/tldraw'
+import {
+	ChangeEvent,
+	KeyboardEvent,
+	ReactNode,
+	SetStateAction,
+	useCallback,
+	useEffect,
+	useLayoutEffect,
+	useRef,
+	useState,
+} from 'react'
+import { FORK_PROJECT_ACTION } from '../../utils/sharing'
+import { SAVE_FILE_COPY_ACTION } from '../../utils/useFileSystem'
+import { getShareUrl } from '../ShareMenu'
+
+type NameState = {
+	readonly name: string | null
+	readonly isEditing: boolean
+}
+
+const MAX_TITLE_WIDTH_PX = 420
+
+export const DocumentTopZone = track(function DocumentTopZone({
+	isOffline,
+}: {
+	isOffline: boolean
+}) {
+	const isDocumentNameVisible = useBreakpoint() >= 4
+
+	return (
+		<DocumentTopZoneContainer>
+			{isDocumentNameVisible && <DocumentNameInner />}
+			{isOffline && <OfflineIndicator />}
+		</DocumentTopZoneContainer>
+	)
+})
+
+export const DocumentNameInner = track(function DocumentNameInner() {
+	const [state, setState] = useState<NameState>({ name: null, isEditing: false })
+	const actions = useActions()
+	const forkAction = actions[FORK_PROJECT_ACTION]
+	const saveFileAction = actions[SAVE_FILE_COPY_ACTION]
+	const editor = useEditor()
+	const msg = useTranslation()
+
+	return (
+		<div className="tlui-document-name__inner">
+			<DocumentNameEditor state={state} setState={setState} />
+			<DropdownMenu.Root id="document name">
```

Comment:
`document name` might be a typo? did you mean to add a '-'? or, is this needed anyway?

Reply:
Think I mostly just replaced `board` with `document` from the original [PR that added board titles in tldraw pro](https://github.com/tldraw/brivate/pull/2972/files#diff-d825ed9304ac39ce1e768f560737448b1af61616404664d42547ed631cd397fcR54). Seems like we used a space delimited name there as well. I'm fine with both 🤷

Reply:
the space weirds me out heh - feels like it shouldn't work :P